core/sequencer/trillian_client.go

```diff
@@ -141,3 +141,34 @@ func (c *MapClient) GetAndVerifyMapRootByRevision(ctx context.Context,
 	}
 	return rawMapRoot, mapRoot, nil
 }
+
+// GetMapLeavesByRevisionNoProof verifies and returns the requested map leaves at a specific revision.
+// indexes may not contain duplicates.
+func (c *MapClient) GetMapLeavesByRevisionNoProof(ctx context.Context, revision int64, indexes [][]byte) ([]*tpb.MapLeaf, error) {
+	if err := hasDuplicates(indexes); err != nil {
+		return nil, err
+	}
+	getResp, err := c.Conn.GetLeavesByRevisionNoProof(ctx, &tpb.GetMapLeavesByRevisionRequest{
+		MapId:    c.MapID,
+		Index:    indexes,
+		Revision: revision,
+	})
+	if err != nil {
+		s := status.Convert(err)
+		return nil, status.Errorf(s.Code(), "map.GetLeaves(): %v", s.Message())
+	}
+	return getResp.Leaves, nil
+}
+
+// hasDuplicates returns an error if there are duplicates in indexes.
```

Comment:
Why is this an error? Checking dups increases cost of every request, but in fact may not be needed cause clients don't send dups. Even if they did (and it was their fault), would it be bad to return 2 copies of the same data?

Reply:
Returning two copies of the same data is not horrible, but it does indicate that something has gone wrong.  This can be removed